osquery/extensions/interface.cpp

```diff
@@ -127,7 +132,20 @@ void ExtensionManagerHandler::registerExtension(
   }
 
   // Every call to registerExtension is assigned a new RouteUUID.
-  RouteUUID uuid = (uint16_t)rand();
+  unsigned int randVal = 0;
+#ifdef WIN32
+  // Windows srand only seeds the current thread, so we make use of the
+  // Windows rand_s function which leverages the OS crypto libs to generate
+  // a cryptographically secure random number. This function has no impact
+  // on srand/rand state, and we fall back to rand() in the event of a rand_s
+  // fail
+  auto ret = rand_s(&randVal);
+  randVal = ret == 0 ? randVal : rand();
```

Comment:
Technically we don't need this to be cryptographically secure, maybe we could get away with `rand()` and seeing in the dispatcher's `run() final` method?